src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -408,16 +453,49 @@ static Value builtinArraySplice(ExecutionState& state, Value thisValue, size_t a
     // If relativeStart is negative, let actualStart be max((len + relativeStart),0); else let actualStart be min(relativeStart, len).
     int64_t actualStart = (relativeStart < 0) ? std::max(len + relativeStart, 0.0) : std::min(relativeStart, (double)len);
 
+    int64_t insertCount;
+    int64_t actualDeleteCount;
+#ifdef ESCARGOT_ENABLE_ES2015
+    // If the number of actual arguments is 0, then
+    if (argc == 0) {
+        // Let insertCount be 0.
+        insertCount = 0;
+        // Let actualDeleteCount be 0.
+        actualDeleteCount = 0;
+    } else if (argc == 1) {
+        // Else if the number of actual arguments is 1, then
+        // Let insertCount be 0.
+        insertCount = 0;
+        // Let actualDeleteCount be len – actualStart.
+        actualDeleteCount = len - actualStart;
+    } else {
+        // Else,
+        // Let insertCount be the number of actual arguments minus 2.
+        insertCount = argc - 2;
+        // Let dc be ToInteger(deleteCount).
+        double dc = argv[1].toInteger(state);
+        // Let actualDeleteCount be min(max(dc,0), len – actualStart).
+        actualDeleteCount = std::min(std::max(dc, 0.0), (double)(len - actualStart));
+    }
+    // If len+insertCount−actualDeleteCount > 2^53-1, throw a TypeError exception.
+    CHECK_ARRAY_LENGTH(len + insertCount - actualDeleteCount, (1LL << 53));
+    // Let A be ArraySpeciesCreate(O, actualDeleteCount).
+    Value val = arraySpeciesCreate(state, O, actualDeleteCount);
+    ASSERT(val.isObject());
+    ASSERT(val.asObject()->isArrayObject());
```

Comment:
For readability / State expression explicitly
ASSERT(val.isObject()->isArrayObject());